crates/sui-indexer/src/handlers/checkpoint_handler.rs

```diff
@@ -125,15 +125,11 @@ where
             .get_checkpoint(seq.into())
             .await?;
 
-        let transactions = join_all(
-            checkpoint
-                .transactions
-                .iter()
-                .map(|tx_digest| self.rpc_client.read_api().get_transaction(*tx_digest)),
-        )
-        .await
-        .into_iter()
-        .collect::<Result<Vec<_>, _>>()?;
+        let transactions = self
```

Comment:
I am a bit worried about the response size getting too large. We will need to limit the response size of `multi_get_transactions` at some point, and it's possible that we may hit the limit with some extra large transaction + checkpoint. Say, we hit 1000TPS * 10 seconds(checkpoint duration) * 1MB(transaction size) = 10GB.

Could you add a `TODO(critical)` comment here and create an issue for this?


Can you add a `TODO` and create an issue that

Reply:
I have this in the PR I am working on , we will need to limit it, the server will return error if the batch size is > 1000

```
  let transactions = join_all(checkpoint.transactions.chunks(MULTI_GET_CHUNK_SIZE).map(
            |digests| {
                self.rpc_client
                    .read_api()
                    .multi_get_transactions(digests.to_vec())
            },
        ))
        .await
        .into_iter()
        .try_fold(vec![], |mut acc, chunk| {
            acc.extend(chunk?);
            Ok::<_, Error>(acc)
        })?;
```

Reply:
@patrickkuo speaking of the input size limit, I am okay with having it as 1000, but we should mostly rely on the final response size check since 1000 full transaction effects might still be too large while 1000 transaction digests(with the default SuiTransactionOptions) is tiny.

Reply:
we will have a response size limit as well on the server, currently it's 1GB, we need to set it to something sensible

Reply:
Good point, I think if we do not have a txn size limit, it is difficult to use txn count to limit the data size here considering extremely big txn like change of epoch or genesis.

Reply:
merging now and will leave the limit to Patrick's PR.